src/python/zquantum/core/cost_function.py

```diff
@@ -338,6 +347,9 @@ def create_cost_function(
         parameter_preprocessors: a list of callable functions that are applied to
             parameters prior to estimation task evaluation. These functions have to
             adhere to the ParameterPreprocessor protocol.
+        gradient_function: a function which returns a function used to compute the
```

Comment:
Why did you add `gradient_function` to this function?

Reply:
To keep it consistent with other cost function factories bc other ones all take a `gradient_function` arg - do you think that's a good idea?

Reply:
Ok, makes sense!